### PR TITLE
add flow chart for messages handled by eventbus

### LINKED ISSUE #100: Add documentation for edge core modules
This is an umbrella issue to describe how each beehive module in kubeedge works.
The expected output is a document which details the responsibilities of the module , its interfaces with other modules etc.

### PR DESCRIPTION
**Pull request by diaojuan**
Fixes #100

### CODE REVIEW

**Comment by m1093782566** [comment]
@diaojuan Thanks for contributing but could you please provide a link to these great images?

**Comment by rohitsardesai83** [comment]
@diaojuan , @m1093782566 , the images look great , however can we add some documentation under "modules" which describes the content of these images ?

**Comment by sids-b** [comment]
> @diaojuan , @m1093782566 , the images look great , however can we add some documentation under "modules" which describes the content of these images ?

Agree with @rohitsardesai83 . Can we have a document with these images describing the flow ??

File: docs/modules/eventbus.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# eventbus
+## **Overview** 
+Eventbus acts as an interface for sending/receiving messages on mqtt topics. It supports 3 kinds of mode: internalMqttMode,externalMqttMode and BothMqttMode.
```
Comment (m1093782566) [inline docs/modules/eventbus.md]: nit: `internalMqttMode`, `externalMqttMode` and `BothMqttMode` for a better look?

File: docs/modules/eventbus.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+## eventbus suscribes to the following topics
+    $hw/events/upload/#
+    SYS/dis/upload_records
+    SYS/dis/upload_records/+
+    $hw/event/node/+/membership/get
+    $hw/event/node/+/membership/get/+
+    $hw/events/device/+/state/update
+    $hw/events/device/+/state/update/+
+    $hw/event/device/+/twin/+
+Note: topic wildcards
```
Comment (m1093782566) [inline docs/modules/eventbus.md]: Suggest:

> Note: topic wildcards
>   #: It must be the last charater in the topic, and matches the current tree and all subtrees.
>   +: It matches exactly one item in the topic tree.

**Comment by m1093782566** [comment]
@pavan187 

Please take a look here, seems dead lock still exists in the integration test?

**Comment by pavan187** [comment]
@m1093782566 , seems multiple PR's have been raised at same time....circleci would be an issue in allocating jobs or containers ..

@sids-b  can you please re-run and confirm ?

File: docs/modules/eventbus.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# eventbus
+## **Overview** 
+Eventbus acts as an interface for sending/receiving messages on mqtt topics. It supports 3 kinds of mode: internalMqttMode,externalMqttMode and BothMqttMode.
+
+## eventbus suscribes to the following topics
```
Comment (sids-b) [inline docs/modules/eventbus.md]: nit: suscribes -> subscribes

File: docs/modules/eventbus.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    SYS/dis/upload_records
+    SYS/dis/upload_records/+
+    $hw/event/node/+/membership/get
+    $hw/event/node/+/membership/get/+
+    $hw/events/device/+/state/update
+    $hw/events/device/+/state/update/+
+    $hw/event/device/+/twin/+
+Note: topic wildcards
+
+    #: It must be the last charater in the topic, and matches the current tree and all subtrees.
```
Comment (sids-b) [inline docs/modules/eventbus.md]: nit: charater -> character

File: docs/modules/eventbus.md (hunk 1 above)
Comment (rohitsardesai83) [reply docs/modules/eventbus.md]: better we can have a list to mention the modes : 
- internalMqttMode
- externalMqttMode
- bothMqttMode

File: docs/modules/eventbus.md (hunk 3 above)
Comment (rohitsardesai83) [inline docs/modules/eventbus.md]: The heading could be "Topics"

and then below that , we can have this line . 
also the topics can be put in a  list format.

File: docs/modules/eventbus.md (hunk 1 above)
Comment (diaojuan) [reply docs/modules/eventbus.md]: fixed

File: docs/modules/eventbus.md (hunk 2 above)
Comment (diaojuan) [reply docs/modules/eventbus.md]: I use table to make it  clear. fixed.

File: docs/modules/eventbus.md (hunk 3 above)
Comment (diaojuan) [reply docs/modules/eventbus.md]: fixed. thanks

File: docs/modules/eventbus.md (hunk 4 above)
Comment (diaojuan) [reply docs/modules/eventbus.md]: fixed,thanks

File: docs/modules/eventbus.md (hunk 3 above)
Comment (diaojuan) [reply docs/modules/eventbus.md]: fixed

**Comment by m1093782566** [comment]
We just merged a fix for CI, let's rerun the CI tests to see if it really fixes something.

**Comment by m1093782566** [comment]
@srivatsav123 CI still failed.

Please check the error message here:

#!/bin/bash -eo pipefail
make verify
bash -x hack/verify.sh
+ curl -L https://git.io/vp6lP
+ sh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100  9857  100  9857    0     0  93893      0 --:--:-- --:--:-- --:--:-- 93893
alecthomas/gometalinter info checking GitHub for latest tag
alecthomas/gometalinter info found version: 2.0.12 for v2.0.12/linux/amd64
install: cannot change permissions of ‘/go/bin’: Operation not permitted
+ export PATH=/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/go/src/github.com/kubeedge/kubeedge/bin
+ PATH=/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/go/src/github.com/kubeedge/kubeedge/bin
+ gometalinter --disable-all --enable=gofmt --enable=misspell --exclude=vendor ./...
hack/verify.sh: line 22: gometalinter: command not found
+ '[' 127 '!=' 0 ']'
+ echo 'Please fix the warnings!'
Please fix the warnings!
+ echo 'Run hack/update-gofmt.sh if any warnings in gofmt'
Run hack/update-gofmt.sh if any warnings in gofmt
+ exit 1
Makefile:22: recipe for target 'verify' failed
make: *** [verify] Error 1
Exited with code 2

**Comment by srivatsav123** [comment]
Hi Dujun,
It seems my fix is not merged (not present in log which you have sent) .if
my fix is already merged please close and reopen the MR.

Regards,
Manju

**Comment by srivatsav123** [comment]
@Dujun
OK sure..

**Comment by m1093782566** [comment]
Per @sids-b 

We may need to rebase to the HEAD once CI changes. 

@diaojuan 

Would you please rebase this PR to the HEAD?

**Comment by m1093782566** [comment]
Per @sids-b

We may need to rebase to the HEAD once CI changes.

@diaojuan

Would you please rebase this PR to the HEAD?

**Pull request merged**